client/src/components/App.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'React';
+import Header from './Header.jsx';
+import Modal from './Modal.jsx';
+import Navbar from './Navbar.jsx';
+import Description from './Description.jsx';
+import Footer from './Footer.jsx';
+
+import $ from 'jquery';
+
+class App extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = { 
+      restaurant: null,
+      currentImage: 2,
+      images: ['https://i.imgur.com/i2FffSR.jpg', 'https://i.imgur.com/rd9yHKH.jpg', 'https://i.imgur.com/ocPGEcz.jpg', 'https://i.imgur.com/noBxGLl.jpg', 'https://i.imgur.com/yajIWRO.jpg']
+    }
+  }
+
+  componentDidMount(){
+    this.getData()
+  }
+
+  getData() {
+    $.ajax({
+      url: '/restaurant/overview',
+      method: 'GET',
+      dataType: 'json',
+      success: (data) => {
+      console.log('success', data)
+      this.setState({
+        restaurant: data[0]
+      })
+    },
+      error: () => {console.log('error')}
+    })    
+  }
+
+  handleModal(e) {
+    console.log('Clicked', e.target.id)
+    if(e.target.id !== "bookmark"){
+    document.getElementById('myModal').style.display = "block";
+    this.setState({
+      currentImage: 0
+    })
+    }
+    if (e.target.id === "closeModal"){
+      document.getElementById('myModal').style.display = "none";
+    }
+    if (e.target.id === "previousImage"){
+      console.log('Previous')
+      if (this.state.currentImage === 0){
+        this.setState({
+          currentImage: this.state.images.length - 1
+        })  
+      } else {      
+        this.setState({
+        currentImage: this.state.currentImage - 1
+      })}
+    }
+    if (e.target.id === "nextImage"){
+      console.log('Next')
+      console.log(this.state.currentImage, this.state.images.length - 1)
+      if (this.state.currentImage === this.state.images.length - 1){
+        this.setState({
+          currentImage: 0
+        })
+      } else {
+      this.setState({
+        currentImage: this.state.currentImage + 1
+      })}
+    }
+  }
+
+
+
+  render() {
+    return (
+      <div>
+      {this.state.restaurant !== null ? 
+      <div>
+        <Header handleModal={this.handleModal.bind(this)} />
+        <Modal handleModal={this.handleModal.bind(this)} state={this.state}/>
+        <div className="overview-container">
+          <div className="overview">
+            <Navbar />
+            <Description restaurant={this.state.restaurant}/>
+          </div>
+        </div>
+        <Footer />
+      </div>
+      : null }
+      </div>
+    )
+  }
+}
+
+$(document).ready(function(){
```

Comment:
This is also a React Anti-pattern. Try to avoid using jquery and react to do DOM manipulation. Only use react